pkg/services/sqlstore/migrations/user_mig.go

```diff
@@ -140,6 +140,19 @@ func addUserMigrations(mg *Migrator) {
 			SQLite(migSQLITEisServiceAccountNullable).
 			Postgres("ALTER TABLE `user` ALTER COLUMN is_service_account DROP NOT NULL;").
 			Mysql("ALTER TABLE user MODIFY is_service_account BOOLEAN DEFAULT 0;"))
+
+	mg.AddMigration("Add uid column to user", NewAddColumnMigration(userV2, &Column{
+		Name: "uid", Type: DB_NVarchar, Length: 40, Nullable: true,
+	}))
+
+	mg.AddMigration("Update uid column values for users", NewRawSQLMigration("").
+		SQLite("UPDATE user SET uid=printf('i%09d',id) WHERE uid IS NULL;").
```

Comment:
better ideas for a prefix?  u?

Eventually this will land in the service account like:
`user:i000034`, or `api-key:i000024`

Reply:
is this not going to expose the internal id's of users? meaning you could get information about how many users in an instance and so forth

@jmatosgrafana you have any comments on this?

Reply:
🤷🏻 that is what we currently do so it won't be worse than that.

It would be nicer if we could fill these in with values from our uid generator, but this lets us do the update in SQL, for sure open to better options!

Reply:
This kind of id is not a secret: if it leaks it doesn't really help to achieve authentication bypass and there are other ways to guess user ids (e.g. changing your email address)
That is why we put enumeration [out of scope in our bug bounty program.](https://github.com/grafana/bugbounty)